CMakeLists.txt

```diff
@@ -136,8 +136,7 @@ option(
 set(AVIF_USE_CXX OFF)
 
 if(AVIF_BUILD_APPS
-   OR AVIF_ENABLE_FUZZTEST
-   OR AVIF_ENABLE_GTEST
+   OR AVIF_BUILD_TESTS
```

Comment:
Thanks for the pull request. Not all the tests in libavif use C++. The original test `AVIF_ENABLE_FUZZTEST OR AVIF_ENABLE_GTEST` is intended to check whether the tests that use C++ are enabled.

However, your pull request points us in the right direction. I found that at this point, we have not defined the options `AVIF_ENABLE_FUZZTEST` and `AVIF_ENABLE_GTEST`. Also, `AVIF_ENABLE_GTEST` should only be checked when `AVIF_ENABLE_GTEST`. So we should test `(AVIF_BUILD_TESTS AND (AVIF_ENABLE_FUZZTEST OR AVIF_ENABLE_GTEST))`, and I think we need to define the options `AVIF_ENABLE_FUZZTEST` and `AVIF_ENABLE_GTEST` earlier, before we test them.

Reply:
@wantehchang 
`AVIF_ENABLE_FUZZTEST` has no effect on `repro_avif_decode_fuzzer`.
This made `AVIF_USE_CXX` mandatory once #1914 was merged.

Reply:
There are indeed a few checks that got mixed up. I believe https://github.com/AOMediaCodec/libavif/pull/2302 fixes them all.

Reply:
Moving the options was not necessary for `AVIF_ENABLE_FUZZTEST` and `AVIF_ENABLE_GTEST` because as having them not set is equivalent to OFF. `AVIF_ENABLE_GTEST` needed to be moved though and I moved other ones for clarity

Reply:
@Jamim , can you please confirm https://github.com/AOMediaCodec/libavif/pull/2302 fixes it for you? Thx.